### PR TITLE
Make The Compiling for Web documentation more obvious

### PR DESCRIPTION
**Pull request by nylnook**
I updated the documentation with workaround if it's not compiling from my experience in Bug #11197, and thanks to the help of @eska014.

### CODE REVIEW

File: development/compiling/compiling_for_web.rst [hunk 1]
```diff
@@ -10,21 +10,31 @@ Requirements
 
 To compile export templates for the Web, the following is required:
 
--  `Emscripten SDK <http://emscripten.org/>`__ (Install in a path without
-   spaces, i.e. not on "Program Files")
--  `Python 2.7+ <https://www.python.org/>`__ (3.0 is
-   untested as of now)
+-  `Emscripten <http://emscripten.org/>`__, plus `LLVM <https://llvm.org/>`__ and `Binaryen <https://github.com/WebAssembly/binaryen>`__ if you want to compile for *Webassembly* (Godot 3). The easiest way to install them if they are not up to date in your Linux package manager is to use the `Emscripten SDK <http://emscripten.org/>`__ (Install in a path without spaces, i.e. not on "Program Files")
```
Comment (eska014) [inline development/compiling/compiling_for_web.rst]: LLVM is not needed. Emscripten comes with its own fork of LLVM called *fastcomp*, which has the *JSBackend* needed for asm.js and asm2wasm.
Binaryen doesn't usually need to be installed manually, this is just a current upstream issue.
So just mention Emscripten.

File: development/compiling/compiling_for_web.rst [hunk 2]
```diff
@@ -10,21 +10,31 @@ Requirements
...
 -  `SCons <http://www.scons.org>`__ build system
 
 Building export templates
 -------------------------
 
 Start a terminal and set the environment variable ``EMSCRIPTEN_ROOT`` to the
-installation directory of Emscripten::
+installation directory of Emscripten:
+
+-  If you installed Emscripten via the Emscripten SDK, you will declare the variable with a path to your downloaded folder:
```
Comment (eska014) [inline development/compiling/compiling_for_web.rst]: This colon needs to be a double colon, so the following paragraph is displayed with a fixed-width typeface.

File: development/compiling/compiling_for_web.rst [hunk 3]
```diff
@@ -10,21 +10,31 @@ Requirements
...
 
     export EMSCRIPTEN_ROOT=~/emsdk/emscripten/master
+    
+-  If you installed Emscripten via your package manager, you can know the path with the ``em-config`` command
+
+    em-config EMSCRIPTEN_ROOT
```
Comment (eska014) [inline development/compiling/compiling_for_web.rst]: Missing double colon before this command line example

File: development/compiling/compiling_for_web.rst [hunk 4]
```diff
@@ -10,21 +10,31 @@ Requirements
...
+
+   So you will declare the variable as for example:
 
-If you are on Windows, start a regular prompt or the Emscripten Command Prompt.
+    export EMSCRIPTEN_ROOT=/usr/lib/emscripten
```
Comment (eska014) [inline development/compiling/compiling_for_web.rst]: Missing double colon before this command line example

File: development/compiling/compiling_for_web.rst [hunk 5]
```diff
@@ -96,6 +108,8 @@ These commands will build WebAssembly export templates in either release or
 debug mode. The generated files' names contain ``.webassembly`` as an
 additional file suffix before the extension.
 
+If it's not compiling, check if you have properly installed everything, and if ``LLVM_ROOT`` and ``BINARYEN_ROOT`` variables are defined in the ``~/.emscripten`` file. You can define the ``BINARYEN_ROOT`` path manually in this file if you already installed it and want to avoid Scons to compile it from ports during template compilation.
```
Comment (eska014) [inline development/compiling/compiling_for_web.rst]: *SCons* is written with an upper-case *C*

File: development/compiling/compiling_for_web.rst (hunk 1 above)
Comment (eska014) [inline development/compiling/compiling_for_web.rst]: You don't need to mention that this applies to Godot 3, since Godot 2 has its own branch in the docs.

Comment (eska014) [inline development/compiling/compiling_for_web.rst]: WebAssembly is written with an upper-case *A*

**Comment by nylnook** [comment]
Thanks for your feedback @eska014, here is a better and corrected version, with no syntax mistakes as far as I can check.

**Comment by NathanLovato** [comment]
Merging now, thanks much for the PR. For future contributions, please check the [writing guidelines](http://docs.godotengine.org/en/latest/community/contributing/docs_writing_guidelines.html)

**Comment by nylnook** [comment]
Thanks, noted ! I'm not a native english speaker, the writing guidelines will be useful.

**Pull request merged**